### ISSUE TITLE
Generic type resolving broken in typescript@next

### ISSUE THREAD

**Issue by awerlogus** [microsoft/TypeScript#44453]
# Bug Report

### 💻 Code

```js
/** @template T @typedef {T | undefined} Option */

/** @type {<T1, T2>(x: Option<T1>, y: T2) => T1 | T2} */
const a = (x, y) => x !== undefined ? x : y

// Making 'number | undefined' value
/** @type {{ prop?: number }} */
const b = { prop: 1 }
const { prop } = b

// 4.3: number
// 4.4: number | undefined
const c = a(prop, 2)

```

### 🙁 Actual behavior

`undefined` is a part of generic `T1`

### 🙂 Expected behavior

`undefined` is excluded from generic `T1`

**Comment by RyanCavanaugh** [comment]
I triaged a duplicate of this *yesterday* and can't find it, so tagging this as bug 😬. It repros in non-JS Doc type space as well

**Comment by ahejlsberg** [comment]
@awerlogus I'm unable to repro this issue. Can you let us know exactly what your config settings are?

**Comment by ahejlsberg** [comment]
@awerlogus Actually, it does repro, just not in the playground for some reason. It is fixed by #44565 and basically a duplicate of #44388, so I will close it as such.
